pkg/plugin/toolregistry/toolregistrytest/toolregistrytest.go

```diff
@@ -38,97 +44,76 @@ type templateValues struct {
 	Os      string
 }
 
-func NewToolRegistry(t *testing.T) (*ToolRegistry, error) {
-	tmpDir, err := os.MkdirTemp("", "tool-registry-test")
-	if err != nil {
-		return nil, err
-	}
-	return &ToolRegistry{
-		testingT: t,
-		tmpDir:   tmpDir,
-	}, nil
-}
-
-func (r *ToolRegistry) newTmpDir() (string, error) {
-	return os.MkdirTemp(r.tmpDir, "")
+type fakeClient struct {
+	pipedservicetest.MockPluginServiceClient
```

Comment:
Use it to reduce the cost of maintaining fake implementation.

Reply:
[IMO]
Using gomock only for this embedding is a bit complex.
How about embedding service.PluginServiceClient instead of generated mock?
https://github.com/pipe-cd/pipecd/blob/a252de8ea9a09e75978472008d9ac3178ad033f1/pkg/plugin/pipedservice/service_grpc.pb.go#L24

Reply:
Thanks, sounds good!
I thought we needed the actual instance to override the specific method. 🙏 
It works fine.

[1d8bea3](https://github.com/pipe-cd/pipecd/pull/5627/commits/1d8bea39927570ebc7f43ddad7378a052004b1a3)